### DISCUSSION TITLE
Alias Commands

### DISCUSSION THREAD

**Discussion by tryoxiss** [helix-editor/helix#12083] [Ideas] [answered]
it would be really nice to have some sort of way to alias typeable commands. For example, I like staying in my editor, but running shell commands in it is a bit annoying for something that is so common to a workflow. It would be nice if I could set a typeable command for commits/linting/rebuilding a program (e.g., running `:runrs` = `cargo run` instead of having a seperate terminal for that or doing `:sh cargo run` all the time). Something along the lines of:

> Sidenote: Actually writing that, it could possibly also work in `languages.toml`, so you could have like `[languages.rust]` which has `commands."run" = [":sh cargo run --quiet"]` so its language dependent. I dont know im tired.

```toml
[commands]
":git-add" = [":sh git add ."]

# Args gets dynamically replaced with the following items, so
# :git-commit -m "Hello"
# will run the helix command
# "git commit -m "Hello"
":git-commit" = [":sh git commit {ARGS}"]

# Other than {ARGS} or a similar mechanism (if it exists, id be happy with just adding new aliases) this uses an identical syntax to
# the [keys] table, so you could do something like this (not that you'd want to do this specifically, but I am sure there are good uses for that and it saves coding a new syntax and users learning a new syntax)

":up" = ["extend_to_line_bounds", "delete_selection", "move_line_up", "paste_before", "move_line_up"]
```

Maybe this already exists, but if it does I wasnt able to find it after searching through the docs.

**Comment by RoloEdits** [comment, marked as answer]
This feature is being tracked here: https://github.com/helix-editor/helix/issues/4423
